addon/components/_private/rdfa-editor/external-triple-form.gts

```diff
@@ -0,0 +1,365 @@
+  };
+  initAfterInsert = modifier((form: HTMLFormElement) => {
+    const formData = new FormData(form);
+    this.currentFormData = formData;
+  });
+  <template>
+    <form
+      ...attributes
+      {{on 'submit' this.handleSubmit}}
+      {{on 'input' this.handleInput}}
+      {{this.initAfterInsert}}
+    >
+
+      <StringField
+        @name='subject.value'
+        @required={{true}}
+        @errors={{this.errors}}
+        @value={{this.triple.subject.value}}
+      >Subject</StringField>
+      <StringField
+        @name='predicate'
+        @required={{true}}
+        @errors={{this.errors}}
+        @value={{this.triple.predicate}}
+      >Predicate</StringField>
+      <TermTypeSelectField
+        @name='termType'
+        @selected={{this.termType}}
+        @options={{this.termTypes}}
+        @onChange={{this.selectTermType}}
+        @errors={{this.errors}}
+        @required={{true}}
+      >
+        TermType
+      </TermTypeSelectField>
+      <StringField
+        @name='object.value'
+        @required={{true}}
+        @errors={{this.errors}}
+        @value={{this.triple.object.value}}
+      >Value</StringField>
+
+      {{#unless (eq this.termType 'NamedNode')}}
+        <StringField
+          @name={{this.datatypePath}}
+          @required={{false}}
+          @errors={{this.errors}}
+          @value={{this.datatype}}
+          @disabled={{this.hasLanguage}}
+        >Datatype</StringField>
+        <StringField
+          @name={{this.languagePath}}
+          @required={{false}}
+          @errors={{this.errors}}
+          @value={{this.language}}
+          @disabled={{this.hasDatatype}}
+        >Language</StringField>
+      {{/unless}}
+    </form>
+  </template>
+}
+interface FieldArgs {
+  name: string;
+  errors: ValidationError[];
+  required: boolean;
+}
+interface StringFieldSig {
+  Blocks: {
+    default: [];
+  };
+  Args: FieldArgs & { value: string; disabled?: boolean };
+}
+
+const StringField: TemplateOnlyComponent<StringFieldSig> = <template>
+  <FormField @name={{@name}} @errors={{@errors}} @required={{@required}}>
+    <:label>
+      {{yield}}
+    </:label>
+
+    <:default as |id|>
+      <AuInput
+        id={{id}}
+        name={{@name}}
+        value={{@value}}
+        required={{@required}}
+        @disabled={{@disabled}}
+        @width='block'
+      />
+    </:default>
+  </FormField>
+</template>;
+
+interface SelectFieldArgs<T> extends FieldArgs {
+  options: T[];
+  selected: T;
+  onChange: (newValue: T, select: Select, event?: Event) => void;
+}
+interface SelectFieldSig<T> {
+  Args: SelectFieldArgs<T>;
+  Blocks: { default: [] };
+}
+
+// TOCs can't have a generic argument, so in this case we have to make a backing
+// class
+// ref: https://typed-ember.gitbook.io/glint/environments/ember/template-only-components
+class SelectField<T> extends Component<SelectFieldSig<T>> {
+  onChange(selection: unknown, select: Select, event?: Event) {
```

Comment:
This needs to either be an arrow function or an `@action`, otherwise `this.args` is undefined.